src/core/renderLayer.js

```diff
@@ -41,7 +44,66 @@ export default class RenderLayer {
     }
 
     getFeaturesAtPosition(pos) {
-        return [].concat(...this.getActiveDataframes().map(df => df.getFeaturesAtPosition(pos, this.style)));
+        return [].concat(...this.getActiveDataframes().map(df => df.getFeaturesAtPosition(pos, this.style))).map(feature => {
+
+            const genReset = styleProperty =>
+                (duration = 500) => {
+                    if (this.styledFeatures[feature.id][styleProperty]) {
+                        this.styledFeatures[feature.id][styleProperty].replaceChild(
+                            this.styledFeatures[feature.id][styleProperty].mix,
+                            // animate(0) is used to ensure that blend._predraw() "GC" collects it
+                            blend(notEquals(property('cartodb_id'), feature.id), animate(0), animate(duration))
+                        );
+                        this.styledFeatures[feature.id][styleProperty] = undefined;
+                    }
+                };
+
+            const genStyleProperty = styleProperty => {
+                const blender = (newExpression, duration = 500) => {
+                    if (typeof newExpression == 'string') {
+                        newExpression = parseStyleExpression(newExpression);
+                    }
+                    if (this.styledFeatures[feature.id] && this.styledFeatures[feature.id][styleProperty]){
+                        this.styledFeatures[feature.id][styleProperty].a.blendTo(newExpression, duration);
+                        return;
+                    }
+                    const blendExpr = blend(
+                        newExpression,
+                        this.style._styleSpec[styleProperty],
+                        blend(1, notEquals(property('cartodb_id'), feature.id), animate(duration))
+                    );
+                    this.trackFeatureStyle(feature.id, styleProperty, blendExpr);
+                    this.style.replaceChild(
+                        this.style.getColor(),
```

Comment:
getColor()? What about the rest of properties?

Reply:
LOL, really good catch